src/jetpack-editor-setup.js

```diff
@@ -144,6 +144,28 @@ const setupHooks = () => {
 	} );
 };
 
+const setupStringsOverrides = () => {
+	addFilter(
+		'native.missing_block_detail',
+		'native/missing_block',
+		( defaultValue, blockName ) => {
+			const { getSettings } = select( 'core/block-editor' );
+			const onlyCoreBlocks =
+				getSettings( 'capabilities' ).onlyCoreBlocks === true;
+
+			const jetpackBlockNames = Object.keys( supportedJetpackBlocks ).map(
+				( name ) => `jetpack/${ name }`
+			);
+
+			if ( onlyCoreBlocks && jetpackBlockNames.includes( blockName ) ) {
+				return null;
+			}
```

Comment:
The default string for the `missing_block_detail` states that it will support the block in the future ([reference](https://github.com/WordPress/gutenberg/blob/afc8b428853937e1ee0f423b346558adbb649502/packages/block-library/src/missing/edit.native.js#L154)), however, the Jetpack blocks won't if we force the editor to only register Core blocks (e.g. when the Jetpack-powered features are disabled).